velero-plugins/migpod/stage.go

```diff
@@ -0,0 +1,71 @@
+package migpod
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/fusor/openshift-migration-plugin/velero-plugins/migcommon"
+	"github.com/sirupsen/logrus"
+	corev1API "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func buildStagePod(pod *corev1API.Pod, log logrus.FieldLogger) *corev1API.Pod {
+	log.Infof("[pod-restore] creating stage pod for %s", pod.Name)
+
+	// Map of Restic volumes.
+	resticVolumes := map[string]bool{}
+	for _, name := range strings.Split(pod.Annotations[migcommon.MigrateQuiesceAnnotation], ",") {
+		resticVolumes[name] = true
+	}
+	// Base pod.
+	newPod := corev1API.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       pod.Kind,
+			APIVersion: pod.APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: pod.Namespace,
+			Name:      pod.Name + "-" + "stage",
+			Annotations: map[string]string{
+				migcommon.MigrateQuiesceAnnotation: pod.Annotations[migcommon.MigrateQuiesceAnnotation],
```

Comment:
In the previous controller code, this referenced `ResticPvBackupAnnotation: pod.Annotations[ResticPvBackupAnnotation]` -- Is this possibly the wrong annotation here? We need the restic annotation so that restic will know to restore volumes for the pod, but I don't think we care about quiesce here, unless there's something else I'm missing.